File: dynamic_listings/lua/default/computer_science/dfs_inorder.lua
local function dfs_traverse_inorder(n)
    -- Step 1: We traverse the left subtree, using recursion
    if (n.left ~= nil) then
        dfs_traverse_inorder(n.left)
    end
    -- Step 2: Visit the node, in this case we print its value
    print(n.content)
    -- Step 3: We traverse the right subtre, using recursion
    if (n.right ~= nil) then
        dfs_traverse_inorder(n.right)
    end
end

local function main()
    local root = build_example_tree()
    dfs_traverse_inorder(root)
end

File: dynamic_listings/javascript/default/computer_science/dfs_inorder.js
function dfs_traverse_inorder(n){
    // Step 1: We traverse the left subtree, using recursion;
    if (n.left != null){
        dfs_traverse_inorder(n.left);
    }
    // Step 2: Visit the node, in this case we print its value
    console.log(n.content);
    // Step 3: We traverse the right subtre, using recursion;
    if (n.right != null){
        dfs_traverse_inorder(n.right);
    }
}

function main(){
    let root = build_example_tree();
    dfs_traverse_inorder(root);
}
